src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -45,10 +46,24 @@ public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(Configura
 
         public bool CanProcess(ConfigurationSetting setting)
         {
-            string contentType = setting?.ContentType?.Split(';')[0].Trim();
+            if (setting == null ||
+                string.IsNullOrWhiteSpace(setting.Value) ||
+                string.IsNullOrWhiteSpace(setting.ContentType))
+            {
+                return false;
+            }
+
+            if (setting.Key.StartsWith(FeatureManagementConstants.FeatureFlagMarker))
+            {
+                return true;
+            }
 
-            return string.Equals(contentType, FeatureManagementConstants.ContentType) ||
-                                 setting.Key.StartsWith(FeatureManagementConstants.FeatureFlagMarker);
+            if (setting.ContentType.TryParseContentType(out ContentType contentType))
```

Comment:
This can be simplified to

```
return setting.ContentType.TryParseContentType(out ContentType contentType) &&
    contentType.IsFeatureFlag();
```